src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: Color,
+    protected a: number,
+    protected b: number,
+    protected c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('One or more sides of the triangle equals 0, try again');
+    }
+
+    [this.a, this.b, this.c] = [a, b, c].sort((sideA, sideB) => sideB - sideA);
+
+    if (this.a >= this.b + this.c) {
+      throw new Error('Not possible to construct a triangle');
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+    const semiperimeter = (a + b + c) / 2;
 
+    const areaOfTriangle = Math.sqrt(semiperimeter
+      * (semiperimeter - a)
+      * (semiperimeter - b)
+      * (semiperimeter - c));
+
+    return Math.floor(areaOfTriangle * 100) / 100;
```

Comment:
Please next time don't ignore mentor's comments and create function for round down to hundredths
![image](https://github.com/mate-academy/ts_figures_classes/assets/54936044/679f7f70-a919-4880-8c3e-2d1824d71261)